src/Simplify.cpp

```diff
@@ -455,6 +458,15 @@ class Simplify : public IRMutator {
             expr = a;
         } else if (is_zero(a)) {
             expr = b;
+        } else if (equal(a, b)) {
+            // x + x = x*2
+            expr = mutate(a * make_const(op->type, 2));
+        } else if (call_a &&
+                   call_a->is_intrinsic(Call::signed_integer_overflow)) {
```

Comment:
This seems common enough that maybe we should have an "is_signed_integer_overflow" helper function here in Simplify.cpp.